// Copyright 2016 The Chromium Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

syntax = "proto2";

option optimize_for = LITE_RUNTIME;

package webapk;

// Creates a WebAPK on the server and returns URL to download WebAPK from Google
// Play.
message CreateWebApkRequest {
  optional WebApk webapk = 1;
}

// Response to CreateWebApkRequest.
message CreateWebApkResponse {
  // Package name to install WebAPK at.
  optional string webapk_package_name = 1;

  // URL to download WebAPK.
  optional string signed_download_url = 2;
}

message WebApk {
  // The URL of the Web App Manifest.
  optional string manifest_url = 2;

  // Chrome's package name.
  optional string requester_application_package = 4;

  // Chrome's version.
  optional string requester_application_version = 5;

  // The Web App Manifest.
  optional WebAppManifest manifest = 6;

  reserved 1, 3;
}

// Contains data from the Web App Manifest.
message WebAppManifest {
  optional string name = 1;
  optional string short_name = 2;
  optional string start_url = 4;
  repeated string scopes = 5;
  repeated Image icons = 6;
  optional string orientation = 9;
  optional string display_mode = 10;
  optional string theme_color = 11;
  optional string background_color = 12;

  reserved 3, 7, 8, 13, 14;
}

message Image {
  // Image's URL.
  optional string src = 1;

  // Murmur2 hash of the icon's bytes. There should not be any transformations
  // applied to the icon's bytes prior to taking the Murmur2 hash.
  optional uint64 hash = 5;

  // Actual bytes of the image. This image may be re-encoded from the original
  // image and may not match the murmur2 hash field above.
  optional bytes image_data = 6;

  reserved 2 to 4;
}
